private/buf/bufcheck/internal/bufcheckserver/internal/bufcheckserverhandle/lint.go

```diff
@@ -190,6 +197,1022 @@ func handleLintDirectorySamePackage(
+		bufprotosource.File.SwiftPrefixLocation,
+		"swift_prefix",
+	)
+}
+
+func handleLintPackageSameOptionValue(
+	responseWriter bufcheckserverutil.ResponseWriter,
+	pkg string,
+	pkgFiles []bufprotosource.File,
+	getFileOptionValue func(bufprotosource.File) string,
+	getFileOptionLocation func(bufprotosource.File) bufprotosource.Location,
+	name string,
+) error {
+	optionValueMap := make(map[string]struct{})
+	for _, file := range pkgFiles {
+		optionValueMap[getFileOptionValue(file)] = struct{}{}
+	}
+	if len(optionValueMap) > 1 {
+		_, noOptionValue := optionValueMap[""]
+		delete(optionValueMap, "")
+		optionValues := slicesext.MapKeysToSortedSlice(optionValueMap)
+		for _, file := range pkgFiles {
+			if noOptionValue {
+				responseWriter.AddProtosourceAnnotation(
+					getFileOptionLocation(file),
+					nil,
+					"Files in package %q have both values %q and no value for option %q and all values must be equal.",
+					pkg,
+					strings.Join(optionValues, ","),
+					name,
+				)
+			} else {
+				responseWriter.AddProtosourceAnnotation(
+					getFileOptionLocation(file),
+					nil,
+					"Files in package %q have multiple values %q for option %q and all values must be equal.",
+					pkg,
+					strings.Join(optionValues, ","),
+					name,
+				)
+			}
+		}
+	}
+	return nil
+}
+
+// HandleLintPackageVersionSuffix is a handle function.
+var HandleLintPackageVersionSuffix = bufcheckserverutil.NewLintFileRuleHandler(handleLintPackageVersionSuffix)
+
+func handleLintPackageVersionSuffix(
+	responseWriter bufcheckserverutil.ResponseWriter,
+	_ bufcheckserverutil.Request,
+	file bufprotosource.File,
+) error {
+	pkg := file.Package()
+	if pkg == "" {
+		return nil
+	}
+	if _, ok := protoversion.NewPackageVersionForPackage(pkg); !ok {
+		responseWriter.AddProtosourceAnnotation(
+			file.PackageLocation(),
+			nil,
+			`Package name %q should be suffixed with a correctly formed version, such as %q.`,
+			pkg,
+			pkg+".v1",
+		)
+	}
+	return nil
+}
+
+// HandleLintProtovalidate is a handle function.
+var HandleLintProtovalidate = bufcheckserverutil.NewMultiHandler(
+	bufcheckserverutil.NewLintMessageRuleHandler(handleLintMessageProtovalidate),
+	bufcheckserverutil.NewLintFieldRuleHandler(handleLintFieldProtovalidate),
+	// NOTE: Oneofs also have protovalidate support, but they
+	//       only have a "required" field, so nothing to lint.
+)
+
+func handleLintMessageProtovalidate(
+	responseWriter bufcheckserverutil.ResponseWriter,
+	_ bufcheckserverutil.Request,
+	message bufprotosource.Message,
+) error {
+	add := func(
+		_ bufprotosource.Descriptor,
+		location bufprotosource.Location,
+		_ []bufprotosource.Location,
+		format string,
+		args ...interface{},
+	) {
+		responseWriter.AddProtosourceAnnotation(
+			location,
+			nil,
+			format,
+			args...,
+		)
+	}
+	// TODO: make buflintvalidate.CheckMessage's signature the same as this function,
+	// instead of accepting a legacy addFunc as an argument. Same comment for CheckField.
```

Comment:
Wrapping the response writer in an addFunc for now, so that we can be certain that the `buflintvalidate` package isn't changed at all, with `diff -r  private/buf/bufcheck/internal/bufcheckserver/internal/buflintvalidate  private/bufpkg/bufcheck/buflint/internal/buflintvalidate` showing no diff.